pkg/cmd/roachtest/mixed_version_jobs.go

```diff
@@ -56,15 +56,13 @@ func (s *backgroundStepper) launch(ctx context.Context, t test.Test, u *versionU
 	if nodes == nil {
 		nodes = u.c.All()
 	}
-	s.m = newMonitor(ctx, t, u.c, nodes)
-	_, s.m.cancel = context.WithCancel(ctx)
+	s.m = u.c.NewMonitor(ctx, t, nodes)
 	s.m.Go(func(ctx context.Context) error {
 		return s.run(ctx, u)
 	})
 }
 
 func (s *backgroundStepper) wait(ctx context.Context, t test.Test, u *versionUpgradeTest) {
-	s.m.cancel()
```

Comment:
>I think the intention was to cancel the context the task is running under, but that is not what is
> happening.

I don't think that was the intention, I'm guessing it was to make `m.cancel` a noop such that an error in any single goroutine would not cancel the other goroutines but let them run to completion. The call in `wait()` is simply to avoid a context leak in the happy case.

However, I traced the callers and the only substantial user that spawns multiple goroutines does so using a new monitor:

https://github.com/cockroachdb/cockroach/blob/8507369d24c96c33b40cd92102228bbd6e8cfa49/pkg/cmd/roachtest/tpcc.go#L219-L223

I guess that makes this ineffectual after all, since that monitor will now have its own child context and cancellation function that would end up cancelling them all on the first error anyway. So it's probably fine, but worth keep this in mind in case we missed anything.

Reply:
I'll leave this open to clean up separately.